subprojects/file-watching/src/test/groovy/org/gradle/internal/watch/registry/impl/AbstractFileWatcherUpdaterTest.groovy

```diff
@@ -378,6 +385,47 @@ abstract class AbstractFileWatcherUpdaterTest extends Specification {
         0 * _
     }
 
+    def "watchers are stopped when watched hierarchy is moved"() {
```

Comment:
Should we add one for moving the parent of the watched hierarchy, too?

Reply:
I have an integration test for moving the parent. I think that is good enough. I don't see what would be different when moving the parent here, given that we mock out the watcher.